### PR TITLE
cephadm-clients playbook to copy keyring with the same name as specified by the keyring parameter

### PR DESCRIPTION
**Pull request by Ameenasuhani**
Closed loop for BZ-2028628

# Description
Execution of cephadm-clients playbook to copy keyring with the same name as specified by the keyring parameter(custom name)

Polarion: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574415
Jira: https://issues.redhat.com/browse/RHCEPHQE-4085

### CODE REVIEW

**Comment by MohitBis** [comment]
@Ameenasuhani 
Testing Logs ?

**Comment by psathyan** [comment]
@Ameenasuhani please attach the logs...

**Comment by Ameenasuhani** [comment]
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-UJDVBR/

**Comment by vamahaja** [comment]
What is BZ-2028628 ? Please provide additional details about code.

**Comment by Ameenasuhani** [comment]
@vamahaja Updated the description

**Review by pranavprakash20 — DISMISSED** [review]
LGTM

**Review by vamahaja — COMMENTED** [review]
@Ameenasuhani what about other test suites which are using same testcase e.g `suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-rpm.yaml`? are they PASSED on test run with these changes ?

**Comment by Ameenasuhani** [comment]
@vamahaja the upgrades would not be impacted as the changes are referring to bootstrap and non-ga`ed deployment(not cdn)
i can attach dummy run logs of deployment from some suites if you think it is necessary

**Comment by vamahaja** [comment]
> @vamahaja the upgrades would not be impacted as the changes are referring to bootstrap and non-ga`ed deployment(not cdn) i can attach dummy run logs of deployment from some suites if you think it is necessary

I would suggest to run tests as it uses same test module.

File: ceph/ceph_admin/bootstrap.py [hunk 1]
```diff
@@ -194,7 +196,9 @@ def bootstrap(self: CephAdmProtocol, config: Dict):
         elif custom_repo:
             self.set_tool_repo(repo=custom_repo)
         else:
-            self.set_tool_repo()
+            repos = ["Tools"]
+            for node in self.cluster.get_nodes():
+                setup_repos(node, base_url=base_url, repos=repos)
```
Comment (psathyan) [inline ceph/ceph_admin/bootstrap.py:201]: Is this something that we can do in a separate PR?

Comment (Ameenasuhani) [reply ceph/ceph_admin/bootstrap.py:201]: This pr needs this changes as we are seeing failure with gpgcheck
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-2GYFRW/Cephadm-ansible_clients_playbook_0.log

Comment (Ameenasuhani) [reply ceph/ceph_admin/bootstrap.py:201]: "yum-config-manager --setopt=gpgcheck=0 --add-repo <url>" cannot be used in a single command and has to be split into two
1.yum-config-manager --add-repo <url>
2. yum-config-manager --setopt=<name>.gpgcheck=0
As  set_tool_repo() is already existing method it also needs modification

**Comment by Ameenasuhani** [comment]
As per offline discussion with @psathyan  will raise a new pr for setup_repos changes and will create a new suites file which is specific to cephadm-clients.yml to maintain the modularity of other tier-1 suites

**Comment by Ameenasuhani** [comment]
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KIUWG3/

File: suites/pacific/cephadm/tier-1_cephadm-clients.yaml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+               address: true
+               labels: apply-all-labels
+               nodes:
+                 - node4
+      abort-on-fail: true
+  - test:
+      name: Service deployment with spec
+      desc: Add services using spec file.
+      module: test_cephadm.py
+      polarion-id: CEPH-83573746,CEPH-83574727
```
Comment (psathyan) [inline suites/pacific/cephadm/tier-1_cephadm-clients.yaml:94]: We are trying to avoid multiple test case ids as the framework cannot pinpoint which suite has failed. Hence, could we revisit this one?

File: suites/pacific/cephadm/tier-1_cephadm-clients.yaml [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        - config:
+            command: shell
+            args:                 # sleep to get all services deployed
+              - sleep
+              - "300"
+  - test:
+      name: Configure client
+      desc: Configure client on node5
+      module: test_client.py
+      polarion-id:
```
Comment (psathyan) [inline suites/pacific/cephadm/tier-1_cephadm-clients.yaml:146]: Please add a Polarion ID

**Review by psathyan — COMMENTED** [review]
Thanks for splitting this into a new suite. Only concern is with polarion test result hygiene

File: suites/pacific/cephadm/tier-1_cephadm-clients.yaml (hunk 3 above)
Comment (Ameenasuhani) [reply suites/pacific/cephadm/tier-1_cephadm-clients.yaml]: @psathyan Done

**Review by psathyan — COMMENTED** [review]
Looks good to me

**Review by psathyan — APPROVED** [review]
